### PR TITLE
Repeated java installation fixed in geoweaver dmg installer

### PR DESCRIPTION
**Pull request by saivivek116**
fixed repeated java installation
fixed password setup retry
redirected messages to log file

### CODE REVIEW

File: docs/geoweaver-dmg-script.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## Geoweaver Startup Script Documentation
```
Comment (ZihengSun) [inline docs/geoweaver-dmg-script.md:1]: Please mark this as developer only.

File: macos-deployment/make-app.sh [hunk 2]
```diff
@@ -26,9 +26,11 @@ export LC_ALL=en_US.UTF-8
 export LANG=en_US.UTF-8
 export LANGUAGE=en_US.UTF-8
 
+CONFIG_FILE="\$HOME/geoweaver/.geoweaver_config"
```
Comment (ZihengSun) [inline macos-deployment/make-app.sh:29]: What if the geoweaver folder doesn't exist? You should create the folder if it doesn't exist.

**Review by ZihengSun — APPROVED** [review]
LGTM, please change the documentation

File: macos-deployment/make-app.sh (hunk 2 above)
Comment (saivivek116) [reply macos-deployment/make-app.sh:29]: It is being created at the line number 190

**Comment by saivivek116** [comment]
Tested with old h2 format. Successfully moving old h2 files into ~h2/old/..

**Pull request merged**